src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,81 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  onAddMovie: (movie: Movie) => void;
+};
+
+const defaultMovie: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const FindMovie: React.FC<Props> = ({ onAddMovie }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [isResponseError, setIsResponseError] = useState(false);
+  const [movie, setMovie] = useState<Movie>(defaultMovie);
+  const [isInitialized, setIsInitialized] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setIsResponseError(false);
+  };
+
+  const onFindClick = async (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+
+    setIsLoading(true);
+
+    try {
+      const response: MovieData | ResponseError = await getMovie(inputValue);
+
+      if ('Error' in response) {
+        setIsResponseError(true);
+        setIsLoading(false);
+
+        return;
+      }
+
+      const movieImdbUrl = `https://www.imdb.com/title/${response.imdbID}`;
+
+      const movieFromServer = {
+        title: response.Title,
+        description: response.Plot,
+        imgUrl: response.Poster !== 'N/A'
+          ? response.Poster
+          : 'https://via.placeholder.com/360x270.png?text=no%20preview',
+        imdbUrl: movieImdbUrl,
+        imdbId: response.imdbID,
+      };
+
+      setIsLoading(false);
```

Comment:
You're setting it in finally so no need to repeat it here.

Reply:
+